### PR TITLE
Ensure vault access only takes place from the controller

### PR DESCRIPTION
**Pull request by jamesrobson-secondmind**
Removes the `SecretRetrieve` class that allowed agents to access vault.

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/com/datapipe/jenkins/vault/credentials/common/VaultHelper.java [hunk 1]
```diff
@@ -76,100 +110,20 @@ static String getVaultSecretKey(@NonNull String secretPath,
...
+                    "The credential id was not configured - please specify the credentials to use.");
         } else {
             LOGGER.log(Level.INFO, "Retrieving vault credential ID : " + id);
         }
         List<VaultCredential> credentials = CredentialsProvider
-            .lookupCredentials(VaultCredential.class,
-                itemGroup,
-                ACL.SYSTEM,
-                Collections.<DomainRequirement>emptyList());
+                .lookupCredentials(VaultCredential.class,
```
Comment (jglick) [inline src/main/java/com/datapipe/jenkins/vault/credentials/common/VaultHelper.java:121]: (whitespace-only changes)

File: src/main/java/com/datapipe/jenkins/vault/credentials/common/VaultHelper.java [hunk 2]
```diff
@@ -37,36 +33,74 @@ static Map<String, String> getVaultSecret(@NonNull String secretPath,
...
+
+            VaultCredential vaultCredential = configuration.getVaultCredential();
+            if (vaultCredential == null)
+                vaultCredential = retrieveVaultCredentials(
+                        configuration.getVaultCredentialId(), context);
+
+            VaultAccessor vaultAccessor = new VaultAccessor(vaultConfig, vaultCredential);
+            vaultAccessor.setMaxRetries(configuration.getMaxRetries());
+            vaultAccessor.setRetryIntervalMilliseconds(
+                    configuration.getRetryIntervalMilliseconds());
```
Comment (jglick) [inline src/main/java/com/datapipe/jenkins/vault/credentials/common/VaultHelper.java:79]: (ditto, except not ignorable by GitHub)

**Review by jglick — APPROVED** [review]
Looks right to the extent I understand this plugin.

File: src/main/java/com/datapipe/jenkins/vault/credentials/common/VaultHelper.java (hunk 1 above)
Comment (jetersen) [reply src/main/java/com/datapipe/jenkins/vault/credentials/common/VaultHelper.java:121]: Continuation indent size according to codestyle is set to `4` and this seems like a `8`
https://github.com/jenkinsci/hashicorp-vault-plugin/blob/0bffd1b067452734d4aca451eeeff2d890f2ce61/.idea/codeStyles/Project.xml#L6

Comment (jglick) [reply src/main/java/com/datapipe/jenkins/vault/credentials/common/VaultHelper.java:121]: I do not care at all what the value is, just noting as an aside that the diff is a bit harder to read due to gratuitous hunks.

**Review by jetersen — APPROVED** [review]

**Pull request merged**